### PR TITLE
Lesson classification

### PR DESCRIPTION
**Pull request by Ansper11**

### CODE REVIEW

**Comment by Ansper11** [comment]
This PR contains:

- Part of the classification lesson plan
- the code of dataset generation

File: Lesson_Classification/Lesson Plan.txt [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+•	Learn how to use Python Skit-learn Library to perform KNN.
+
+
+Background:
+
+Emeralds are very popular commodities among the villagers, so you decide to collect them through mining. 
+However, your robot can only mine one mine due to fuel problems, and you don’t know what ore is in each mine. 
+In this world, different kinds of ore will affect the growth of plants on the ground, so if you can find patterns in that,
+You can predict what ore is going to be in each mine. you could use your agent to collect plants from the mine and use the 
+data the company has collected to predict what ore the mine contains. Here’s some data collected by some miners.
```
Comment (Nick-McGee) [inline Lesson_Classification/Lesson Plan.txt:24]: Is there a way where the user can create their own datasets? Jeff seems very interested in user's doing this themselves.

**Comment by Ansper11** [comment]
For the training set, they will use our data because it's really time-consuming if we ask users to collect the whole training dataset. But after the training, we could ask them to collect some new data in the game to verify the performance of the model

File: Lesson_Classification/Lesson Plan.txt (hunk 1 above)
Comment (Ansper11) [reply Lesson_Classification/Lesson Plan.txt:24]: they can collect a small validation set, but not a training set

File: Lesson_Classification/Lesson Plan.txt [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+•	Learn how to use python matplotlib library to print and analyse data.
+
+•	Learn how to calculate the distance between vectors (points)
+
+•	Learn Normalization
+
+•	Implement K-nearest neighbors by python.
+
+•	Learn how to use Python Skit-learn Library to perform KNN.
+
```
Comment (eloiseespel) [inline Lesson_Classification/Lesson Plan.txt:16]: I love this idea of classification, it is  such a visual concept. I think that would really grab the students' attention

**Comment by yyht6300a** [comment]
Be careful. It's sometimes hard to build a world scene in MC. You have to think about what you want the agent to do and how it will connect with the knowledge.
